generic_packer_template.json.j2

```diff
@@ -31,9 +31,30 @@
     "provisioners": [
       {% for provisioner in event['provisioners'] %}
         {
-          "type": "{{ provisioner['type'] or 'shell' }}",
-          "script": "{{ provisioner['script'] }}"
+                  "type": "shell",
```

Comment:
You've now taken a completely generic mechanism and made it be kafka-ami-building specific.  You need to get the thing that generates your Lambda payload file to populate the provisioner portion of the event JSON correctly.  The trick here is that https://github.com/dwp/ami-builder/blob/master/ami_builder.py#L50-L60 copies all of the scripts you put in the config S3 bucket to /tmp inside the Lambda.  You should then be able to put a bunch of file-provisioner configs in the event payload to have them copied from /tmp to wherever they need to be on the EC2 instance that packer is using to build the AMI from.  A bunch of script-provisioner configs in the event payload can then set permissions, etc and I think you'll be good to go.

Reply:
The provisioners section does require more work to become generic, for instance the current format will accept only type "shell", however it is necessary at least another type , that is "file" with different keys (i.e. not "script" for instance).
This should be addressed in another iteration.